proto-public/pbmesh/v1alpha1/http_route.proto

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1;
+
+import "pbmesh/v1alpha1/common.proto";
+import "pbmesh/v1alpha1/http_route_retries.proto";
+import "pbmesh/v1alpha1/http_route_timeouts.proto";
+
+// NOTE: this should align to the GAMMA/gateway-api version, or at least be
+// easily translatable.
+//
+// https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRoute
```

Comment:
This is not a blocker at all, but I thought I'll ask here. Should we think about how to run GAMMA conformance tests against these to make sure we stay compliant or even to check if these are compliant? Maybe something we should track somewhere as a deliverable